internal/terraform/eval_conditions.go

```diff
@@ -91,24 +99,46 @@ func evalCheckRules(typ checkType, rules []*configs.CheckRule, ctx EvalContext,
 			diags = diags.Append(&hcl.Diagnostic{
 				Severity:    hcl.DiagError,
 				Summary:     errInvalidCondition,
-				Detail:      fmt.Sprintf("Invalid validation condition result value: %s.", tfdiags.FormatError(err)),
+				Detail:      fmt.Sprintf("Invalid condition result value: %s.", tfdiags.FormatError(err)),
 				Subject:     rule.Condition.Range().Ptr(),
 				Expression:  rule.Condition,
 				EvalContext: hclCtx,
 			})
 			continue
 		}
 
-		if result.False() {
-			diags = diags.Append(&hcl.Diagnostic{
-				Severity:    hcl.DiagError,
-				Summary:     typ.FailureSummary(),
-				Detail:      rule.ErrorMessage,
-				Subject:     rule.Condition.Range().Ptr(),
-				Expression:  rule.Condition,
-				EvalContext: hclCtx,
-			})
+		if result.True() {
+			continue
+		}
+
+		var errorMessage string
+		if !errorDiags.HasErrors() && errorValue.IsKnown() && !errorValue.IsNull() {
+			var err error
+			errorValue, err = convert.Convert(errorValue, cty.String)
+			if err != nil {
+				diags = diags.Append(&hcl.Diagnostic{
+					Severity:    hcl.DiagError,
+					Summary:     "Invalid error message",
+					Detail:      fmt.Sprintf("Error message must be a string: %s.", tfdiags.FormatError(err)),
+					Subject:     rule.ErrorMessage.Range().Ptr(),
+					Expression:  rule.ErrorMessage,
+					EvalContext: hclCtx,
+				})
+			} else {
+				errorMessage = errorValue.AsString()
+			}
+		}
+		if errorMessage == "" {
+			errorMessage = "Condition error message could not be evaluated."
```

Comment:
Passive voice nit :grimacing:: Perhaps "Could not evaluate condition error message."? (Or, as we sometimes do in other spots, "Failed to ...")

Reply:
Oops, thanks! Passive voice keeps happening.